### PR TITLE
Fixed setServiceLink validation

### PR DESCRIPTION
**Pull request by alena1108**
https://github.com/rancherio/rancher/issues/1215
https://github.com/rancherio/rancher/issues/1216

### CODE REVIEW

**Comment by alena1108** [comment]
update_label_test failed again...not related to the fixes. @sonchang we should look at what else might be fixed to get rid of this failure

**Comment by alena1108** [comment]
@ibuildthecloud

File: code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/api/action/SetServiceLinksActionHandler.java [hunk 1]
```diff
@@ -120,14 +123,19 @@ private void removeOldServiceMaps(Service service, Map<Long, String> newServiceL
             } else {
                 String newName = newServiceLinks.get(existingMap.getConsumedServiceId());
                 String existingName = existingMap.getName();
-                if (newName != null && existingName != null && !newName.equals(existingName)) {
+
+                boolean bothNull = (newName == null && existingName == null);
+                boolean oneNull = !bothNull && (newName == null || existingName == null);
```
Comment (ibuildthecloud) [inline code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/api/action/SetServiceLinksActionHandler.java]: Are you just doing ObjectUtils.equals()?

Comment (alena1108) [inline code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/api/action/SetServiceLinksActionHandler.java]: Ah, StringUtils.equalsIgnoreCase. Will change

**Pull request merged**